akka-stream-tests/src/test/java/akka/stream/stage/JavaIdentityStage.java

```diff
@@ -0,0 +1,45 @@
+package akka.stream.stage;
+
+import akka.stream.Attributes;
+import akka.stream.FlowShape;
+import akka.stream.Inlet;
+import akka.stream.Outlet;
+
+public class JavaIdentityStage<T> extends GraphStage<FlowShape<T, T>> {
+  private Inlet<T> _in = Inlet.of("Identity.in");
+  private Outlet<T> _out = Outlet.of("Identity.out");
+  private FlowShape<T, T> _shape = FlowShape.of(_in, _out);
+
+  public Inlet<T> in() { return _in; }
+  public Outlet<T> out() { return _out; }
+
+
+  @Override
+  public GraphStageLogic createLogic(Attributes inheritedAttributes) {
```

Comment:
Even though it is Java, this stage looks almost as compact as the Scala version.